x-pack/plugins/reporting/server/export_types/png_v2/create_job.ts

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { cryptoFactory } from '../../lib';
+import { CreateJobFn, CreateJobFnFactory } from '../../types';
+import { JobParamsPNGV2, TaskPayloadPNGV2 } from './types';
+
+export const createJobFnFactory: CreateJobFnFactory<
+  CreateJobFn<JobParamsPNGV2, TaskPayloadPNGV2>
+> = function createJobFactoryFn(reporting, logger) {
+  const config = reporting.getConfig();
+  const crypto = cryptoFactory(config.get('encryptionKey'));
+
+  return async function createJob(
+    { objectType, title, locatorParams, browserTimezone, layout, ...jobParams },
+    context,
+    req
+  ) {
+    const serializedEncryptedHeaders = await crypto.encrypt(req.headers);
+
+    return {
+      ...jobParams,
+      headers: serializedEncryptedHeaders,
+      spaceId: reporting.getSpaceId(req, logger),
+      objectType,
+      title,
+      locatorParams: [locatorParams],
+      browserTimezone,
+      layout,
+      forceNow: new Date().toISOString(),
```

Comment:
Same as https://github.com/elastic/kibana/pull/101048/files#r686260362 , we don't need to destructure all of these job params.